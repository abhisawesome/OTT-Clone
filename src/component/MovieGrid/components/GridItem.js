import { memo,lazy } from "react";
import utils from "../../../utils";
import PropType from 'prop-types';


const Image = lazy(() => import("../../Image"));
const Row = ({ data = [], lastElementRef }) => {
    utils.log('Row rendered !');    
    return (
        <div className="grid sm:grid-cols-3 md:grid-cols-5 lg:grid-cols-7 gap-x-[30px] gap-y-[90px] mx-[30px]">
            {data.map((item, index) => {
                if ((data.length - 10) === index + 1) {
                    return (
                        <div
                            ref={lastElementRef}
                            key={index}
                        >
                            {/* <img
                                alt="poster"
                                src={`images/${item["poster-image"]}`}
                            /> */}
                            <Image
                                alt=""
                                className="md:object-contain  md:max-h-[250px] lg:object-contain lg:max-h-[250px]"
                                src={`images/${item["poster-image"]}`}
                            />
                            <p
                                className="line-clamp-1 mt-[24px]"
                            >
                                {item.name}
                            </p>
                        </div>
                    )
                } else {
                    return (
                        <div
                            key={index}
                        >
                            {/* <img
                                alt="poster"
                                src={`images/${item["poster-image"]}`} /> */}
                            <Image
                                alt=""
                                className="md:object-contain md:max-h-[250px] lg:object-contain lg:max-h-[250px]"
                                src={`images/${item["poster-image"]}`}
                            />
                            <p
                                className="line-clamp-1 mt-[24px]"
                            >{item.name}
                            </p>
                        </div>
                    )
                }

            })}
        </div>

    )
}

Row.propTypes = {
    data: PropType.array,
    lastElementRef: PropType.any,
}
export default memo(Row);