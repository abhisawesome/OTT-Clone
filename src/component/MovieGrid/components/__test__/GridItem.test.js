import { Suspense, useRef } from 'react';
import renderer from 'react-test-renderer';
import GridItem from '../GridItem';
const data = [{
    "name": "The Birds",
    "poster-image": "poster1.jpg"
},
{
    "name": "Rear Window",
    "poster-image": "poster2.jpg"
}]
describe('Grid Item', () => {
    test('Snapshot test', () => {
        let ref = () => { }
        const tree = renderer.create(
            <Suspense fallback={<div>Loading..</div>}>
                <GridItem
                    lastElementRef={ref}
                    data={data}
                />
            </Suspense>
        )
        expect(tree).toMatchSnapshot();
    })
})