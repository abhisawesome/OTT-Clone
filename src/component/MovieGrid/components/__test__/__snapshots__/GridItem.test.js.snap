// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Item Snapshot test 1`] = `
<div>
  Loading..
</div>
`;
